src/rest_api/classes/generic_fields.clj

```diff
@@ -178,3 +178,52 @@
     {:data (not-empty data)
      :description  (str "external databases and IDs containing "
                         "additional information on the object")}))
+
+(defn references [object]
+  (let [id-kw (first (filter #(= (name %) "id") (keys object)))
+        role (namespace id-kw)
+        data (when (some? id-kw)
+               (let [reference-kw (keyword role "reference")]
+                 (when-let [papers (reference-kw object)]
+                   (let [number-of-papers (count papers)
+                         kw-reference-paper (keyword (str role ".reference") "paper")]
+                     {:count number-of-papers
+                      :results (for [ph (if (or (kw-reference-paper papers)
+                                                (:paper/id papers)) [papers] papers)
+                                     :let [paper (if (contains? ph kw-reference-paper)
+                                                   (kw-reference-paper ph)
+                                                   ph)]]
+                                 (let [abstract (:paper/abstract paper)
+                                       publication-date (:paper/publication-date paper)
+                                       pt (:paper/type paper)
+                                       author-holder (:paper/author paper)
+                                       year (if (nil? publication-date) nil (first (str/split publication-date #"-")))]
+                                   {:page (:paper/page paper)
+                                    :volume (:paper/volume paper)
+                                    :name  (pack-obj paper)
+                                    :title  [(:paper/title paper)]
+                                    :author (when-let [hs (:paper/author paper)]
+                                              (vals
+                                                (into
+                                                  (sorted-map)
+                                                  (into
+                                                    {}
+                                                    (for [h hs
+                                                          :let [author (cond
+                                                                         (contains? h :affiliation/person)
+                                                                         (first (:affiliation/person h))
+
+                                                                         (contains? h :paper.author/author)
+                                                                         (:paper.author/author h))]]
+                                                      {(:ordered/index h)
+                                                       (pack-obj author)})))))
+                                    :ptype (if (nil? paper) nil (:paper.type  pt))
+                                    :abstract (if (nil? abstract) nil [(:longtext/text (first abstract))])
```

Comment:
Same comment as above, rather than having to code both clauses of `if`, just have:
```clojure
(when abstract
   [(:longtext/text (fist abstract))])
```